### PR TITLE
add the Elastic Collector distribution

### PR DESCRIPTION
**Pull request by rogercoll**
# Changes

Replace the Contrib collector with the Elastic distribution:

- [x] Docker compose
- [x] K8s

Note that the `elasticsearch exporter` has not been added yet to the configuration pipelines. For the moment, the configuration still uses the Otlp endpoint to forward all the logs, traces and metrics.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [ ] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: src/otelcollector/otelcol-config-extras.yml [hunk 1]
```diff
@@ -16,26 +16,3 @@
 #    pipelines:
 #      traces:
 #        exporters: [spanmetrics, otlphttp/example]
-
```
Comment (rogercoll) [inline src/otelcollector/otelcol-config-extras.yml:19]: Rollback to upstream version

File: kubernetes/elastic-helm/values.yaml [hunk 2]
```diff
@@ -24,38 +30,23 @@ opentelemetry-collector:
...
-          processors: [batch]
-          exporters: [otlp/elastic]
-        logs:
-          receivers: [otlp]
-          processors: [batch]
-          exporters: [otlp/elastic]
+
+  configMap:
+    create: false
+    existingName: elastic-otelcol-agent
```
Comment (rogercoll) [inline kubernetes/elastic-helm/values.yaml:36]: Needed to create a custom configMap as unable to disable components that are not available in our distribution. This is due to a known issue in Helm: https://github.com/helm/helm/issues/12488

File: kubernetes/elastic-helm/values.yaml [hunk 3]
```diff
@@ -4,6 +4,12 @@ default:
     tag: 1.10.1
 
 opentelemetry-collector:
+  image:
+    repository: docker.elastic.co/beats/elastic-agent
+    tag: 8.15.0-7b611e39-SNAPSHOT
```
Comment (ChrsMark) [inline kubernetes/elastic-helm/values.yaml:9]:
```suggestion
    tag: 8.15.0-SNAPSHOT
```
Maybe just point to the latest SNAPSHOT for now? Eventually we should switch to `8.15.0`.

File: kubernetes/elastic-helm/configmap-elastic.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    app.kubernetes.io/name: otelcol
+
+data:
+  relay: |
+    connectors:
+      spanmetrics: {}
+    exporters:
+      debug: {}
+      otlp/elastic:
+        endpoint: ${ELASTIC_APM_ENDPOINT}
```
Comment (ChrsMark) [inline kubernetes/elastic-helm/configmap-elastic.yaml:17]:
```suggestion
        endpoint: ${env:ELASTIC_APM_ENDPOINT}
```
Hmm, isn't the `env:` prefix required here? I'm always confused with that 😄  but further down we have  `${env:MY_POD_IP}`

**Review by ChrsMark — COMMENTED** [review]
Looks good overall, left some minors.

**Pull request merged**